src/lib/dovetail_audio.ts

```diff
@@ -0,0 +1,221 @@
+import {AdzerkRequest, AdzerkResponse} from './adzerk';
+import {AdzerkFetcher} from './adzerk_fetcher';
+import {ExtendableAudio} from '../javascript/extendable_audio';
+import {DovetailFetcher, DovetailResponse, ArrangementEntry} from './dovetail_fetcher';
+
+type AllUnion = [
+  DovetailResponse | PromiseLike<DovetailResponse>,
+  AdzerkResponse | PromiseLike<AdzerkResponse>
+];
+
+type AllResult = [DovetailResponse, AdzerkResponse];
```

Comment:
These 2 types basically help me get around a deficiency with the Typescript handling of `Promise.all()` without being crazy verbose later.